src/ring_theory/dedekind_domain.lean

```diff
@@ -108,17 +108,115 @@ structure is_dedekind_domain_dvr : Prop :=
 (is_dvr_at_nonzero_prime : ∀ P ≠ (⊥ : ideal A), P.is_prime →
   discrete_valuation_ring (localization.at_prime P))
 
+section inverse
+
+open_locale classical
+
+variables {R₁ : Type*} [integral_domain R₁] {g : fraction_map R₁ K}
+
+noncomputable instance fractional_ideal_has_div :
+  has_div (fractional_ideal g) :=
+⟨ λ I J, if h : J = 0 then 0 else ⟨I.1 / J.1, fractional_ideal.fractional_div_of_nonzero h⟩ ⟩
+
+variables {I J : fractional_ideal g} [ J ≠ 0 ]
```

Comment:
Should this be
```suggestion
variables {I J : fractional_ideal g} [fact (J ≠ 0)]
```

Reply:
I have implemented this, although I confess I don't know why should this be better.

I have also resolved all `sorry`'s in `dedekind_domain` but the one concerning `is_dedekind_domain_inv_iff` because this will depende on how we'll define invertible ideals in Dedekind domains.

Reply:
The hypothesis has been removed entirely, so marking this as resolved.